packages/h5p-server/src/ContentManager.ts

```diff
@@ -90,14 +96,27 @@ export default class ContentManager {
     }
 
     /**
-     * Deletes a piece of content and all files dependent on it.
+     * Deletes a piece of content, the corresponding contentUserData and all files dependent on it.
      * @param contentId the piece of content to delete
      * @param user the user who wants to delete it
      */
     public async deleteContent(
         contentId: ContentId,
         user: IUser
     ): Promise<void> {
+        if (this.contentUserDataStorage) {
+            try {
+                await this.contentUserDataStorage.deleteContentUserData(
+                    contentId,
+                    user
```

Comment:
The content user data must be deleted for all users, not just the current.

Reply:
contentUserData is deleted based on the contentId - the user is only passed to check if he has permissions to delete the states.